src/utils.ts

```diff
@@ -14,5 +14,18 @@ export async function graphqlQuery(url: string, query) {
     },
     body: JSON.stringify({ query })
   });
-  return await res.json();
+  let responseData: any = await res.text();
+  try {
+    responseData = JSON.parse(responseData);
+  } catch (e) {
+    throw new Error(`Text response: ${responseData}`);
+  }
+  return responseData;
+}
+
+export function subgraphError(res, error: null | string = null, errors = []) {
+  if (error) return res.status(500).json({ errors: [{ message: error }] });
```

Comment:
Can we return the exact same error than what we get on the subgraph instead of doing our own format?

Reply:
The result might includes others params than just `errors`, I'd rather do something like this:
```ts
return res.status(500).json(response);
```
Also the function `subgraphError` can be quite confusing with both `error` and `errors` for different use case. I think we should either do 2 functions or just use `res.status(500).json(response)` directly in the api.ts file